src/Memory.cs

```diff
@@ -119,10 +217,30 @@ public unsafe Span<T> GetSpan<T>(int address)
         /// <typeparam name="T">Type of the struct to read. Ensure layout in C# is identical to layout in WASM.</typeparam>
         /// <param name="address">The zero-based address to read from.</param>
         /// <returns>Returns the struct read from memory.</returns>
+        /// <remarks>
+        /// Note that WebAssembly always uses little endian as byte order. On platforms 
+        /// that use big endian, you will need to convert numeric values accordingly.
+        /// </remarks>
         public T Read<T>(int address)
             where T : unmanaged
         {
-            return GetSpan<T>(address)[0];
+            return GetSpan<T>(address, 1)[0];
+        }
+
+        /// <summary>
+        /// Read a struct from memory.
+        /// </summary>
+        /// <typeparam name="T">Type of the struct to read. Ensure layout in C# is identical to layout in WASM.</typeparam>
+        /// <param name="address">The zero-based address to read from.</param>
+        /// <returns>Returns the struct read from memory.</returns>
+        /// <remarks>
+        /// Note that WebAssembly always uses little endian as byte order. On platforms 
+        /// that use big endian, you will need to convert numeric values accordingly.
+        /// </remarks>
+        public T Read<T>(long address)
+            where T : unmanaged
+        {
+            return GetSpan<T>(address, 1)[0];
```

Comment:
Could we just remove these many new `long` overloads and change the `int`-based ones to `long` given that `int` implicitly converts to `long`?

Reply:
Yes, we could remove the `int`-based overloads as they are already replaced with `long`-based ones. However, while that would be a source-compatible change (since `int` implicitely converts to `long`), it would be a binary-incompatible change (it requires a recompilation of the assembly that calls the methods).

If `wasmtime-dotnet` uses a similar release cycle as `wasmtime` which always increments the major version, I think such a change should be OK.
Thanks!

Reply:
The `int`-based overloads are now replaced by the `long` based overloads.

Reply:
Agree, I am more concerned with source compatibility given the way Wasmtime will be rev'ing version numbers. Thanks for making these changes.